cassandra/metadata.py

```diff
@@ -458,16 +475,14 @@ class SimpleStrategy(ReplicationStrategy):
     """
 
     def __init__(self, options_map):
-        try:
-            self.replication_factor = int(options_map['replication_factor'])
-        except Exception:
-            raise ValueError("SimpleStrategy requires an integer 'replication_factor' option")
+        self.replication_factor = options_map['replication_factor']
```

Comment:
self.replication_factor is an integer since the beginning... it might be better to not change this. Could you add a `_raw_replication_factor` attribute and preserve `self.replication_factor` for the normal/total replica int value?

Reply:
That seems a little confusing to me. Are you worried about breaking existing behavior? The only place SimpleStrategy can get instantiated (other than tests) is here:
https://github.com/datastax/python-driver/blob/master/cassandra/metadata.py#L691

and nothing in the metadata parsing directly references instance attributes of the RF classes. If `replication_factor` was being accessed directly, you'd need to do a check to make sure you have a SimpleStrategy because other RFs don't have it.

Calling `export_as_string()` already prints it as a string, so no worries there.

Reply:
`replication_strategy` is exposed in the KeyspaceMetadata, as you pointed. That means it is breaking change if the user is using the metadata to do anything. I understand it is not a critical breaking change but it is still one. No matter if they check what type is the strategy class... they expect an integer if they read that attribute.

Another option: Since the transient replication is not enabled by default, I'd would prefer we keep things as is since it's 99.99% of our current users and fallback to a `string` attribute if TR is detected when parsing. We can document this properly in the attribute docs. wdyt?

Reply:
I see. Since backwards compatibility is an issue, I think I prefer your first suggestion then. I thought having `replication_factor` not represent the "true" replication factor would be confusing, but if we have to do this to keep backwards compatibility, that's fine.

Reply:
that was a valid point too. Should we expose a `replication_transient_factor` then?

Reply:
I added a separate `transient_replicas` field. Let me know if that looks good to you.